stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -509,6 +489,79 @@ void TaskGroupImpl::destroy(AsyncTask *task) {
   swift_task_dealloc(task, this);
 }
 
+// =============================================================================
+// ==== add --------------------------------------------------------------------
+
+SWIFT_CC(swiftasync)
+void swift::swift_taskGroup_add(
+    AsyncTask *addingTask,
+    ExecutorRef executor,
+    SWIFT_ASYNC_CONTEXT AsyncContext *rawContext) {
+  addingTask->ResumeTask = rawContext->ResumeParent;
+  addingTask->ResumeContext = rawContext;
+
+  auto context = static_cast<TaskGroupAddAsyncContext *>(rawContext);
+  auto group = context->group;
+  auto _group = asImpl(group);
+  auto operation = context->operation;
+  // to get the effective priority of this task, we check if an override was specified,
+  // and if not we re-use the parent task's priority.
+  auto priority = context->priorityOverride == JobPriority::Unspecified ?
+                  addingTask->Flags.getPriority() : context->priorityOverride;
+  auto childTaskResultType = context->childTaskResultType;
+  fprintf(stderr, "[%s:%d] (%s) params --------------\n", __FILE__, __LINE__, __FUNCTION__);
+  fprintf(stderr, "[%s:%d] (%s) addingTask  %d\n", __FILE__, __LINE__, __FUNCTION__, addingTask);
+  fprintf(stderr, "[%s:%d] (%s) group       %d\n", __FILE__, __LINE__, __FUNCTION__, group);
+  fprintf(stderr, "[%s:%d] (%s) override    %d\n", __FILE__, __LINE__, __FUNCTION__, context->priorityOverride);
+  fprintf(stderr, "[%s:%d] (%s) priority    %d\n", __FILE__, __LINE__, __FUNCTION__, priority);
+  fprintf(stderr, "[%s:%d] (%s) operation   %d\n", __FILE__, __LINE__, __FUNCTION__, operation);
+  assert(group && "swift_taskGroup_wait_next_throwing was passed context without group!");
+
+  // If the task is cancelled, the group is as well,
+  // and we can avoid even trying to add into it
+  auto cancelled = addingTask->isCancelled();
+
+  // Unless already cancelled, attempt to +1 pending task count in the group.
+  // This will also inform us if the group itself already is cancelled.
+  if (!cancelled) {
+    auto assumedStatus = _group->statusAddPendingTaskRelaxed();
+    // if the group was cancelled already,
+    cancelled = cancelled && assumedStatus.isCancelled();
+  }
+
+  // If the group or task were cancelled, we cannot add tasks and must bail out
+  if (cancelled) {
+    // initialize the result as `false` (unable to add to the group).
+    *reinterpret_cast<bool*>(context->successResultPointer) = false;
+    addingTask->runInFullyEstablishedContext(executor);
+    return;
+  }
+
+  // We added `pending` and can create the group child task now
+  auto flags = JobFlags(JobKind::Task, priority);
+  flags.task_setIsFuture(true);
+  flags.task_setIsChildTask(true);
+  flags.task_setIsGroupChildTask(true);
+
+  auto pair = swift_task_create_group_future_f(
+    flags, /*parent*/ addingTask, group,
+    /*futureResultType*/ childTaskResultType,
+    operation->Function.get(),
+    operation->ExpectedContextSize);
```

Comment:
I'm having trouble getting this piece right...

The context size seems to be quite off:

`    frame #6: 0x0000000100160484 libswift_Concurrency.dylib`swift_task_create_group_future_f(flags=JobFlags @ r12, parent=0x0000000101706c30, group=0x0000000109008820, futureResultType=0x0000000108805000, function=(0x00000000edcf7638), initialContextSize=1396790588) at Task.cpp:273 [opt]` it does not seem like a proper value for initialContextSize 🤔 

which then causes crashes in the storageOffset:

```
* thread #2, queue = 'com.apple.root.default-qos', stop reason = EXC_BAD_ACCESS (code=1, address=0x50)
    frame #0: 0x0000000100160488 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::TargetValueWitnessFlags<unsigned long>::getAlignmentMask(this=<unavailable>) const at MetadataValues.h:174:18 [opt]
   171 	  /// For example, if the type needs to be 8-byte aligned, the
   172 	  /// appropriate alignment mask should be 0x7.
   173 	  size_t getAlignmentMask() const {
-> 174 	    return (Data & AlignmentMask);
   175 	  }
   176 	  constexpr TargetValueWitnessFlags withAlignmentMask(size_t alignMask) const {
   177 	    return TargetValueWitnessFlags((Data & ~AlignmentMask) | alignMask);
Target 0: (a.out) stopped.
(lldb) bt
* thread #2, queue = 'com.apple.root.default-qos', stop reason = EXC_BAD_ACCESS (code=1, address=0x50)
  * frame #0: 0x0000000100160488 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::TargetValueWitnessFlags<unsigned long>::getAlignmentMask(this=<unavailable>) const at MetadataValues.h:174:18 [opt]
    frame #1: 0x0000000100160488 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::TargetValueWitnessFlags<unsigned long>::getAlignment(this=<unavailable>) const at MetadataValues.h:180 [opt]
    frame #2: 0x0000000100160488 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::TargetValueWitnessTable<swift::InProcess>::getAlignment(this=0x0000000000000000) const at Metadata.h:391 [opt]
    frame #3: 0x0000000100160488 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::TargetMetadata<swift::InProcess>::vw_alignment(this=0x0000000108805000) const at Metadata.h:649 [opt]
    frame #4: 0x0000000100160484 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::AsyncTask::FutureFragment::storageOffset(resultType=0x0000000108805000) at Task.h:395 [opt]
    frame #5: 0x0000000100160484 libswift_Concurrency.dylib`swift_task_create_group_future_f [inlined] swift::AsyncTask::FutureFragment::fragmentSize(resultType=0x0000000108805000) at Task.h:402 [opt]
    frame #6: 0x0000000100160484 libswift_Concurrency.dylib`swift_task_create_group_future_f(flags=JobFlags @ r12, parent=0x0000000101706c30, group=0x0000000109008820, futureResultType=0x0000000108805000, function=(0x00000000edcf7638), initialContextSize=1396790588) at Task.cpp:273 [opt]
    frame #7: 0x0000000100162013 libswift_Concurrency.dylib`swift_taskGroup_add(addingTask=<unavailable>, executor=(Value = 0), rawContext=<unavailable>) at TaskGroup.cpp:546:15 [opt]
    frame #8: 0x000000010015f0f7 libswift_Concurrency.dylib`swift_job_run [inlined] swift::AsyncTask::runInFullyEstablishedContext(this=<unavailable>, currentExecutor=<unavailable>) at Task.h:193:5 [opt]
    frame #9: 0x000000010015f0ed libswift_Concurrency.dylib`swift_job_run [inlined] swift::runJobInExecutorContext(job=<unavailable>, executor=<unavailable>) at Actor.cpp:157 [opt]
    frame #10: 0x000000010015f0c7 libswift_Concurrency.dylib`swift_job_run(job=<unavailable>, executor=<unavailable>) at Actor.cpp:142 [opt]
```